addons/mail/static/src/models/composer.js

```diff
@@ -140,6 +143,32 @@ registerModel({
                 composerView.update({ hasToRestoreContent: true });
             }
         },
+        async performRPClinkPreview(url) {
+            const linkPreviews = await this.messaging.rpc({
+                route: '/mail/link_preview',
+                params: { url, channel_id: this.activeThread.id },
+            }, { shadow: true });
+            return linkPreviews;
+        },
+        /**
+         * Try to fetch link preview when urls are inserted inside the composer.
+         */
+        async refreshLinkPreview(pastedText) {
+            const urls = pastedText.match(urlRegexp) || false;
+            if (!urls) {
+                return;
+            }
+            const existingUrls = this.linkPreviews.map(att => att.url);
+            const filteredUrls = urls.filter(
+                url => !existingUrls.includes(url) && !this.linkPreviewsRemovedUrls.includes(url)
+            );
+            for (const url of filteredUrls) {
+                const linkPreviews = await this.performRPClinkPreview(url);
+                if (linkPreviews) {
+                    this.update({ linkPreviews });
+                }
```

Comment:
ideally return value should be made so that update is always ok to call rather than requiring a condition